src/OrleansServiceBus/Providers/Streams/EventHub/EventHubCheckpointer.cs

```diff
@@ -3,17 +3,14 @@
 using System.Threading.Tasks;
 using Microsoft.ServiceBus.Messaging;
 using Orleans.AzureUtils;
+using Orleans.Streams;
 
 namespace Orleans.ServiceBus.Providers
 {
     /// <summary>
-    /// This class stores EventHub partition checkpoint information (a partition offset) in azure table storage.
-    /// TODO: Use dependency injection to fill this behavior.
-    /// This class stores data into table storage, which will not be ideal for all users.  Further, it introduces
-    ///   a dependency on azure storage to this assembly.  Ideally this should be an injectable behavior, rather
-    ///   than a concrete class.
+    /// This class stores EventHub partition checkpointer information (a partition offset) in azure table storage.
     /// </summary>
-    internal class EventHubPartitionCheckpoint
+    public class EventHubCheckpointer : IStreamQueueCheckpointer<string>
```

Comment:
The main thing is the data in the checkpoint.  In EventHub, this data is just a string which is the starting offset.  This will likely change in a future iteration, as I suspect the cache high water mark will also be needed, so this component is still in flux.

So the EventHubCheckpointer stores an EventHub checkpoint into azure table storage.  Maybe AzureTableEventHubCheckpointer?

I did some initial prototyping of an AzureTableCheckpointer<T> (in AzureUtils) that could be used by the EventHubPartitionCheckpointer as EventHubPartitionCheckpointer  : AzureTableCheckpointer<string>, but it got a little complicated, and was outside the scope of this feature.  I have the prototype in a branch and may dig into it more in the future, but I'm trying to focus on the critical path.  The primary goal here was to allow service developers to provide their own checkpoint mechanism.  A more significant refactoring would be a good improvement, but is outside the scope of this change.